### ISSUE TITLE
BE: Drupalize "Create Account" block content

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#11692]
**Blocked on CMS Collab Cycle (small lift) – [CMS CC intake ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12062)**

## Description

![image.png](https://images.zenhubusercontent.com/61a671f5fc46c2a311655f75/5d7ee88d-7ada-4821-addb-0f07a0140ad2)

### Implementation details

- See #12019 for guidance on implementation.
- Design was comlpleted in #12388 
- Design is [in Figma](https://www.figma.com/file/OJ6ncc4hAvABO6iX5Z1Aw6/Create-Account-content-block?node-id=46%3A1754&t=iU8VIU4XjCCm5s27-0)

### Testing considerations
- Spect Tool tests

### Questions after CMS Collab Cycle:
- We're making it more reusable: what is it called now?
- Who can create/edit it?
  - Start with Homepage editors (same permissions as other blocks on new homepage)
- Need character limits
- Need final designs

## Acceptance Criteria
- [ ] Create custom block type called "CTA with links"
- [ ] Fields per [Figma](https://www.figma.com/file/OJ6ncc4hAvABO6iX5Z1Aw6/Create-Account-content-block?node-id=46%3A1754&t=iU8VIU4XjCCm5s27-0)
  - [ ] Teaser text – required
  - [ ] CTA Button (grouping)
    - [ ] CTA Button text - required
    - [ ] CTA button URL - required
    - [ ] Only one button can be added
  - [ ] Related info links (grouping, not required) - max 3
    - [ ] URL (optional)
    - [ ] Link text (required if adding a URL)
    - [ ] Multiple links can be added by clicking "Add another link" button, up to the 3 limit max
    - [ ] Helptext should match text provided in the Figma design (with exception to related info link text help text. We elected to defer the related info link text's help text to a [future ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12696) as it cannot be done in the CMS currently without bespoke code)
- [ ] Implement character limits for each of the following fields (field names subject to change):
    - [ ] Teaser content - 100 chars
    - [ ] Button text - 30 chars
    - [ ] ~~Related info link text - 50 chars~~ Deferred requirement to a [future issue](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12696).
- [ ] Create entityqueue for the New Hompage placement
  - [ ] Restrict to one "CTA with links" block (both min and max)
  - [ ] Name EQ "V2 Home page Create Account"
- [x] Provide GQL query for FE
- [x] Update FE ticket #12532 with any needed details (eg GQL) including: treat links as optional
- [ ] Requires CMS design review for consistency with [Figma](https://www.figma.com/file/OJ6ncc4hAvABO6iX5Z1Aw6/Create-Account-content-block?node-id=46%3A1754&t=iU8VIU4XjCCm5s27-0)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by swirtSJW** [comment]
Probably a custom block type is not the way to go for a single instance.
May want to consider using a centralized content node and a graphQL query that grabs that node specifically to populate the FE.

**Comment by wesrowe** [comment]
@DanielleThierryUSDSVA @rlhecht – FYSA, we will need character limits for the visible fields in this block on the New Homepage. Lmk if this ticket is sufficient or you would like an intake ticket or similar. Implementation may occur in the next sprint or the one after.

![image.png](https://images.zenhubusercontent.com/61a671f5fc46c2a311655f75/5d7ee88d-7ada-4821-addb-0f07a0140ad2)

Fields needing limits (field names may change):
    - [ ] Teaser content – indicate future character limits
    - [ ] Button text – indicate future character limits
    - [ ] Related info link text – indicate future character limits

**Comment by DanielleThierryUSDSVA** [comment]
Hi @wesrowe. 

- The current teaser content comes in at 92 characters with spaces. I think we can make the character limit 100 characters.
- Button character limit should be our standard 35 characters.
- We generally aim for <100 characters in text links. But in this space, that feels like it would be too long. So I'm thinking of keeping the limit more aligned with the button length, but slightly longer. Let's go with 50 characters max.

Feel free to reach out with questions, thoughts, etc. Thanks!

**Comment by thejordanwood** [comment]
@dsasser I think it should be "CTA with links" (plural). Changing it on the design now!

**Comment by dsasser** [comment]
@wesrowe @jilladams one thing I just realized is that the 'create account' button on the new home page doesn't actually link to anything (no href). Here is the button code:

```html
<button class="vads-u-padding-x--4 vads-u-margin-bottom--3" id="bdd48157ac8ba8932cc888d3e6fe6784">
              Create account
            </button>
```

When clicking the 'create account' button, javascript triggers opening the login modal (rather than linking to a destination href). I mention this so that we are aware that the destination for the button won't be coming from the CMS on the new home page's implementation of this create account block.

**Comment by dsasser** [comment]
GraphQL query:

```graphql
query homePageCreateAccountQuery {
  entitySubqueueById(id: "v2_home_page_create_account") {
    ... on EntitySubqueueV2HomePageCreateAccount {
      itemsOfEntitySubqueueV2HomePageCreateAccount {
        entity {
          entityId
          ... on BlockContentCtaWithLink {
            entityId
            entityLabel
            fieldCtaSummaryText
            fieldPrimaryCtaButtonText
            # primary cta button url not necessary for new-home-page
            # fieldPrimaryCtaButtonUrl {
            #   url {
            #     path
            #   }
            # }
            fieldRelatedInfoLinks {
              title
              url {
                path
              }
            }
          }
        }
      }
    }
  }
}
```
Example result:

```json
{
  "data": {
    "entitySubqueueById": {
      "itemsOfEntitySubqueueV2HomePageCreateAccount": [
        {
          "entity": {
            "entityId": "226",
            "entityLabel": "Home page create account CTA",
            "fieldCtaSummaryText": "Create an account to manage your VA benefits and care in one place — anytime, from anywhere.",
            "fieldPrimaryCtaButtonText": "Create account",
            "fieldRelatedInfoLinks": [
              {
                "title": "Learn how an account helps you",
                "url": {
                  "path": "/resources/creating-an-account-for-vagov"
                }
              }
            ]
          }
        }
      ]
    }
  }
}
```

**Comment by wesrowe** [comment]
>When clicking the 'create account' button, javascript triggers opening the login modal (rather than linking to a destination href). I mention this so that we are aware that the destination for the button won't be coming from the CMS on the new home page's implementation of this create account block.

@dsasser, I see what you mean. But is the javascript triggering the page refresh? I see the page flash, and the url is updated with query param `?next=loginModal`.

Either way, should we also be making the CTA Button URL field optional? In this case it would not be used.

**Comment by dsasser** [comment]
> @dsasser, I see what you mean. But is the javascript triggering the page refresh? I see the page flash, and the url is updated with query param `?next=loginModal`.

I haven't looked into how the javascript works at all, but I did also notice the page refresh/flash.

> Either way, should we also be making the CTA Button URL field optional? In this case it would not be used.
I'm not sure. In this case, a user can enter "<none>" as the button url which the system will accept. Additionally, we can simply exclude the button url from the graphql query.

**Comment by dsasser** [comment]
FYI the PR adds a new create-account (CTA with Links) block (via deploy hook), and populate the new entityqueue, to make testing easier and deploy smoother.

**Comment by jilladams** [comment]
https://prod.cms.va.gov/admin/structure/block/block-content?info=&type=cta_with_link
![Screenshot 2023-02-24 at 3 00 32 PM](https://user-images.githubusercontent.com/85581471/221316694-7218fcb6-7611-4046-bdc2-269198bc9a89.png)

https://prod.cms.va.gov/block/add?destination=/admin/structure/block/block-content
![Screenshot 2023-02-24 at 3 01 38 PM](https://user-images.githubusercontent.com/85581471/221316792-25d7619d-f9b6-4a52-ae4a-527c2e9912a5.png)

https://prod.cms.va.gov/block/add/cta_with_link?destination=/admin/structure/block/block-content
![screencapture-prod-cms-va-gov-block-add-cta-with-link-2023-02-24-15_02_00](https://user-images.githubusercontent.com/85581471/221316910-13683b26-7fb4-443a-b620-ce2468a0a959.png)

And the actual new Homepage block: https://prod.cms.va.gov/block/227?destination=/admin/structure/block/block-content

Huzzah! Closing, and we can revisit anything FE needs during implementation, as it comes.
